core-api/src/main/java/com/optimizely/ab/config/audience/match/EqualsAttributeCondition.java

```diff
@@ -0,0 +1,85 @@
+/**
+ *
+ *    Copyright 2019, Optimizely and contributors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+package com.optimizely.ab.config.audience.match;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Objects;
+
+/**
+ * Matches input using standard {@link Object#equals(Object)}.
+ *
+ * If an attribute is set to value that is not an instance of {@code inputClass},
+ * this will always evaluate to UNKNOWN.
+ *
+ * @param <T> type of value to match input against
+ */
+public class EqualsAttributeCondition<T, U> extends AbstractAttributeCondition<T, U> {
```

Comment:
It's surprising that a class with this name is only used for exact _boolean_ conditions, and not for exact number or exact string conditions.